src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -834,7 +834,8 @@ private void InitializePackagesFolderInfo(string packagesFolderPath, bool exclud
                 prunedAvailablePackages = PrunePackageTree.PruneByUpdateConstraints(prunedAvailablePackages, projectInstalledPackageReferences, resolutionContext.VersionConstraints);
 
                 // Verify that the target is allowed by packages.config
-                GatherExceptionHelpers.ThrowIfVersionIsDisallowedByPackagesConfig(primaryTargetIds, projectInstalledPackageReferences, prunedAvailablePackages);
+                GatherExceptionHelpers.ThrowIfVersionIsDisallowedByPackagesConfig(primaryTargetIds, projectInstalledPackageReferences, prunedAvailablePackages,
+                    new LoggerAdapter(nuGetProjectContext));
```

Comment:
Not in this case since both are in different methods. But I'll update it to have only one initialization per method.